langs/jbang.yaml

```diff
@@ -0,0 +1,159 @@
+    extendedClientCapabilities:
+      advancedExtractRefactoringSupport: true
+      advancedGenerateAccessorsSupport: true
+      advancedOrganizeImportsSupport: true
+      classFileContentsSupport: true
+      generateConstructorsPromptSupport: true
+      generateToStringPromptSupport: true
+      hashCodeEqualsPromptSupport: true
+      moveRefactoringSupport: true
+      overrideMethodsPromptSupport: true
+      progressReportProvider: true
+    settings:
+      java:
+        autobuild:
+          enabled: true
+        codeGeneration:
+          generateComments: false
+          hashCodeEquals:
+            useInstanceof: false
+            useJava7Objects: false
+          toString:
+            codeStyle: STRING_CONCATENATION
+            limitElements: 0
+            listArrayContents: true
+            skipNullValues: false
+            template: ${object.className} [${member.name()}=${member.value}, ${otherMembers}]
+          useBlocks: false
+        completion:
+          enabled: true
+          favoriteStaticMembers:
+            - org.junit.Assert.*
+            - org.junit.Assume.*
+            - org.junit.jupiter.api.Assertions.*
+            - org.junit.jupiter.api.Assumptions.*
+            - org.junit.jupiter.api.DynamicContainer.*
+            - org.junit.jupiter.api.DynamicTest.*
+            - org.mockito.Mockito.*
+            - org.mockito.ArgumentMatchers.*
+            - org.mockito.Answers.*
+          filteredTypes:
+            - java.awt.*
+            - com.sun.*
+          guessMethodArguments: true
+          importOrder:
+            - java
+            - javax
+            - com
+            - org
+          overwrite: true
+        configuration:
+          checkProjectSettingsExclusions: true
+          updateBuildConfiguration: automatic
+        dependency:
+          packagePresentation: flat
+        errors:
+          incompleteClasspath:
+            severity: warning
+        foldingRange:
+          enabled: true
+        format:
+          comments:
+            enabled: true
+          enabled: true
+          onType:
+            enabled: true
+        implementationsCodeLens:
+          enabled: false
+        import:
+          exclusions:
+            - '**/node_modules/**'
+            - '**/.metadata/**'
+            - '**/archetype-resources/**'
+            - '**/META-INF/maven/**'
+          gradle:
+            enabled: true
+            wrapper:
+              enabled: true
+          maven:
+            enabled: true
+        maven:
+          downloadSources: false
+        maxConcurrentBuilds: 1
+        progressReports:
+          enabled: true
+        referencesCodeLens:
+          enabled: false
+        saveActions:
+          organizeImports: false
+        selection:
+          enabled: true
+        signatureHelp:
+          enabled: true
+        trace:
+          server: "off"
+  code: "TODO"
+  item: "TODO"
```

Comment:
Would you mind including the changes from https://github.com/raxod502/riju/pull/121 to the language server configuration?

Reply:
Sure. 

Btw. Do you know of a way to have the lsp run on the directory output of a command ? Jbang can set up ide workspace that has dependencies etc. Configured.

Reply:
Sure, you'll probably want to do something like for Q# where you also install the language at build time, and use its tooling to generate a project skeleton:

https://github.com/raxod502/riju/blob/85afa783e5532206f66df5120ad0bac6e47eec70/langs/qsharp.yaml#L25-L37

Which can then be populated into the runtime directory at container start:

https://github.com/raxod502/riju/blob/85afa783e5532206f66df5120ad0bac6e47eec70/langs/qsharp.yaml#L43-L46

Reply:
It's more the other way. 

Need to run 'jbang edit --no-open main.java' which scans the java code and sets up the right ide metadata. Can also run with --live to have it continuously update it if java dependencies changes. 

Note it's just relevant for the Lsp so might just be enough do it when user toggle it on/off.

Reply:
I see. Well, if you need to have the user code before starting LSP, then you probably want to put this into the `lsp.start` command, as you suggested.

Reply:
> These seem like they should be part of the same language, instead of being separate. We generally don't have multiple different variants for a language.

jshell is a variance of java used for jshell.

jbang is just wrapping it here.

Can't have it in one language as it requires a different named file (.jsh vs .java).